lib/compareValues.js

```diff
@@ -78,7 +80,9 @@ function compareValues (a, b) {
     case 'NULL':
       return spaceship(a, b)
     case 'PTYPE<TIME>':
-      return spaceship(a.getTime(), b.getTime())
+      const timeA = _.isDate(a) ? a.getTime() : a.epoch_time * 1000
```

Comment:
I would go back to `return spaceship(a.getTime(), b.getTime())`

If we are not using raw, then there is no point in making the computer do the check.